packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,31 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for a post-blob activated chain.
+    /// Params are passed in as part of msg.data in order to compress the calldata.
+    /// Params should be passed in in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _number             L1 blocknumber.
+    ///   5. _timestamp          L1 timestamp.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+        require(msg.sender == DEPOSITOR_ACCOUNT, "L1Block: only the depositor account can set L1 block values");
+        uint256 data;
+        assembly {
+            data := calldataload(4)
+            sstore(basefeeScalar.slot, shr(192, data)) // basefeeScalar & blobBaseFeeScalar (uint32,uint32)
+            sstore(sequenceNumber.slot, shr(128, data)) // uint64
```

Comment:
Catch by @clabby - there are dirty bits being stored in storage in the `sequenceNumber` slot but it doesn't impact correctness because of the way that the storage layout is set up. There is nothing else shared in storage with the `sequenceNumber`. It is probably best to not dirty additional storage (especially in a not so visible way) so we should `shl` first before `shr` to clear the most significant bits. We should add to our test coverage reading directly from storage with `vm.load` in `testFuzz_setL1BlockValues_succeeds` to ensure that the exact values are set in storage that we care about and nothing more. Could do this by casting the fuzzer inputs to `bytes32` and comparing to the output of `vm.load`

Reply:
The alternative would be to accept the dirty bits as a teeny gas optimization and then add an internal spacer variable that fills up the rest of the storage slot that `sequenceNumber` occupies to make it explicit that its used storage

Reply:
I think I prefer cleaning the bits before storing.  It's not a lot of extra gas.